### PR TITLE
improve api doc for core/algorithm

### PR DESCRIPTION
**Pull request by minggo**
Re: https://github.com/cocos/cocos-engine/issues/13888

### Changelog

* improve api doc for core/algorithm

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/algorithm/binary-search.ts [hunk 1]
```diff
@@ -25,26 +25,26 @@
...
- * Searches the **sorted** number array for an element and returns the index of that element.
+ * Searches the **ascending sorted** array for an element and returns the index of that element.
  * @param array The array to search in.
  * @param value The value to search.
- * @return The index of the searched element in the sorted array, if one is found;
- * otherwise, a negative number that is the bitwise complement of the index of the next element that is large than the searched value or,
- * if there is no larger element(include the case that the array is empty), the bitwise complement of array's length.
+ * @return The index of the searched element in the sorted array, if found;
+ * otherwise, returns the complement of the index of the next element larger than the element to be searched or,
+ * returns the complement of array's length if no element is larger than the element to be searched or the array is empty.
```
Comment (shrinktofit) [inline cocos/core/algorithm/binary-search.ts]: greater than

File: cocos/core/algorithm/binary-search.ts [hunk 2]
```diff
@@ -25,26 +25,26 @@
...
 /**
- * Searches the **sorted** number array for an element and returns the index of that element.
+ * Searches the **ascending sorted** array for an element and returns the index of that element.
  * @param array The array to search in.
  * @param value The value to search.
- * @return The index of the searched element in the sorted array, if one is found;
- * otherwise, a negative number that is the bitwise complement of the index of the next element that is large than the searched value or,
- * if there is no larger element(include the case that the array is empty), the bitwise complement of array's length.
+ * @return The index of the searched element in the sorted array, if found;
+ * otherwise, returns the complement of the index of the next element larger than the element to be searched or,
```
Comment (shrinktofit) [inline cocos/core/algorithm/binary-search.ts]: element to be searched -> searching element 好一点

File: cocos/core/algorithm/easing.ts [hunk 3]
```diff
@@ -23,22 +23,48 @@
  THE SOFTWARE.
  */
 
+// Easing functions specify the speed at which an animation progresses at different points within the animation.
+
+/**
+ * @en Not any easing effect.
+ * @zh 没有任何缓动效果
```
Comment (shrinktofit) [inline cocos/core/algorithm/easing.ts]: punctuators

**Review by shrinktofit — APPROVED** [review]

File: cocos/core/algorithm/move.ts [hunk 4]
```diff
@@ -1,28 +1,32 @@
...
+ * @param first @en The beginning of the original range. @zh 移动区间的起始坐标。
+ * @param last @en The end of the original range. @zh 移动区间的结束坐标。
+ * @return @en The input array. @zh 输入的数组。
  * @engineInternal
  */
-export function move<T> (array: T[], index: number, newIndex: number) {
-    assertsArrayIndex(array, index);
-    assertsArrayIndex(array, newIndex);
-    if (index === newIndex) {
+export function shift<T> (array: T[], first: number, last: number) {
```
Comment (dumganhar) [inline cocos/core/algorithm/move.ts:14]: Should this file be renamed to `shift.ts` or `array-shift.ts` ?

File: cocos/core/algorithm/binary-search.ts [hunk 5]
```diff
@@ -65,13 +65,13 @@ export function binarySearchEpsilon (array: Readonly<ArrayLike<number>>, value:
...
 /**
- * Searches the **sorted** array for an element and returns the index of that element.
+ * Searches the **ascending sorted** array for an element and returns the index of that element.
  * @param array The array to search in.
  * @param value The value to search.
- * @param lessThan The predicate which implements the less than semantic.
- * @return The index of the searched element in the sorted array, if one is found;
- * otherwise, a negative number that is the bitwise complement of the index of the next element that is large than the searched value or,
- * if there is no larger element(include the case that the array is empty), the bitwise complement of array's length.
+ * @param lessThan Comparison function object which returns ​true if the first argument is less than the second.
```
Comment (dumganhar) [inline cocos/core/algorithm/binary-search.ts:71]: `Comparison function` -> `The comparison function`

File: cocos/core/algorithm/move.ts (hunk 4 above)
Comment (minggo) [reply cocos/core/algorithm/move.ts:14]: I thought it before. But as i thought, shift can be a special move operation, so i didn't rename it.

File: cocos/core/algorithm/binary-search.ts (hunk 5 above)
Comment (minggo) [reply cocos/core/algorithm/binary-search.ts:71]: I referred to STL documentation: https://en.cppreference.com/w/cpp/algorithm/sort. It doesn't use `the`:
WeChatWorkScreenshot_a716f556-d04b-47a8-9bae-a6198197753c

File: cocos/core/algorithm/easing.ts [hunk 6]
```diff
@@ -47,14 +73,28 @@ export function quadInOut (k: number) {
...
+/**
+ * @en Starts slowly and accelerates. Refer to [this doc](https://docs.cocos.com/creator/manual/zh/tween/tween-function.html) for graphic feature.
+ * @zh 启动慢，加速快。具体效果可以参考[该文档](https://docs.cocos.com/creator/manual/zh/tween/tween-function.html)。
+ */
 export function cubicIn (k: number) {
     return k * k * k;
 }
 
+/**
+ * @en Starts quickly and the decelerates.
```
Comment (dumganhar) [inline cocos/core/algorithm/easing.ts]: `Starts quickly and the decelerates.` -> `Starts quickly and decelerates.`

File: cocos/core/algorithm/easing.ts [hunk 7]
```diff
@@ -79,14 +133,28 @@ export function quartInOut (k: number) {
...
+/**
+ * @en Starts slowly and accelerates. Refer to [this doc](https://docs.cocos.com/creator/manual/zh/tween/tween-function.html) for graphic feature.
+ * @zh 启动慢，加速快。具体效果可以参考[该文档](https://docs.cocos.com/creator/manual/zh/tween/tween-function.html)。
+ */
 export function quintIn (k: number) {
     return k * k * k * k * k;
 }
 
+/**
+ * @en Starts quickly and the decelerates.
```
Comment (dumganhar) [inline cocos/core/algorithm/easing.ts]: Remove `the`.

File: cocos/core/algorithm/easing.ts [hunk 8]
```diff
@@ -95,29 +163,58 @@ export function quintInOut (k: number) {
...
+/**
+ * @en Starts slowly and accelerates. Refer to [this doc](https://docs.cocos.com/creator/manual/zh/tween/tween-function.html) for graphic feature.
+ * @zh 启动慢，加速快。具体效果可以参考[该文档](https://docs.cocos.com/creator/manual/zh/tween/tween-function.html)。
+ */
 export function expoIn (k: number) {
     return k === 0 ? 0 : Math.pow(1024, k - 1);
 }
 
+/**
+ * @en Starts quickly and the decelerates.
```
Comment (dumganhar) [inline cocos/core/algorithm/easing.ts]: Remove `the`.

File: cocos/core/algorithm/move.ts [hunk 9]
```diff
@@ -1,28 +1,32 @@
...
+    assertsArrayIndex(array, first);
+    assertsArrayIndex(array, last);
+    if (first === last) {
         return array;
     }
-    const element = array[index];
-    if (index < newIndex) { // Shift right
-        for (let iElement = index + 1; iElement <= newIndex; ++iElement) {
+    const element = array[first];
+    if (first < last) { // Shift right
```
Comment (dumganhar) [inline cocos/core/algorithm/move.ts]: The comment says :
```
If `first < last`, then will do left circle shift;
```
But it's actually a right shift here?

**Review by dumganhar — APPROVED** [review]

**Pull request merged**